npm/ipsm/ipsm.go

```diff
@@ -96,6 +96,7 @@ func (ipsMgr *IpsetManager) CreateList(listName string) error {
 	log.Logf("Creating List: %+v", entry)
 	if errCode, err := ipsMgr.Run(entry); err != nil && errCode != 1 {
 		log.Errorf("Error: failed to create ipset list %s.", listName)
+		metrics.SendErrorMetric(util.IpsmID, "Error: failed to create ipset list %s.", listName)
```

Comment:
Can we use `fmt.Sprintf` to format a single string error message and use that string for both `log.Errorf()` and `metrics.SendErrorMetric()`? I think that would keep consistency between sources

Reply:
Sure, I can do that.

Reply:
I think we can even go a step further :). In order to keep things DRY, it seems you can just create or wrap an error, bubble it as its already being done, and move all the logging and metrics to the caller. That would avoid having to log and send a metric on every error in this section. Let me know if that makes sense.

Reply:
For wrapping log and metrics, yes, I do think in same way to avoid duplicated logic.